### ISSUE TITLE
Continued Errors After HA Restart

### ISSUE THREAD

**Issue by kksligh** [JurajNyiri/HomeAssistant-Tapo-Control#472]
### Description

After restarting, this integration throws errors for one of the cameras.  Oddly enough, it is a bit random on which one it is.  I have 3 Tapo cameras connected....

Camera 1
-  Model: C211
-  Firmware: 1.3.9 Build 231019 Rel.31045n(5553)

Camera 2 
-  Model: C211
-  Firmware: 1.3.9 Build 231019 Rel.31045n(5553)

Camera 3
-  Model: C225
-  Firmware: 1.0.27 Build 231120 Rel.55235n

Camera 3 was recently updated to a new firmware and it seems to be the most stable.  Camera's 1 and 2 randomly will throw an error and it impacts the motion detection, ,motion alarm and noise detection functionality when it does error.  They become unavailable.

### Reproduction Steps

Restart HA and it happens nearly every time.

### Expected behavior

Restart HA and it not error out.  

### If applicable, add error logs.

[debug logs.txt](https://github.com/JurajNyiri/HomeAssistant-Tapo-Control/files/13446020/debug.logs.txt)

### Device Firmware

Firmware: 1.3.9 Build 231019 Rel.31045n(5553)

### Integration Version

5.4.12

### Using stream component

Yes

### Does camera work via official integrations?

Yes

### Camera has all attributes filled out in developer tools

Yes

### HASS Environment

HA OS installed on a NUC with Google Coral TPU

### Search for similar issues

Yes

**Comment by kranzhoff** [comment]
Same issue here. 
I have 4 x Tapo C310 cameras setup and the availability of "binary_sensor.[…]_motion_alarm" and "binary_sensor.[…]_noise" of each camera seems to be totally random.
Looking at the history chart of these sensors for a couple of days, they disappear for 6-10 hours and sometimes come back for just a few minutes to a few hours to then disappear again.

The camera settings and network setup did not change all the time.
Not sure if this is related to restarts of HASS,  but I restart HASS quite often these days.

Here are the relevant states from developer tools => states if this helps:

[binary_sensor_states.txt](https://github.com/JurajNyiri/HomeAssistant-Tapo-Control/files/13452188/binary_sensor_states.txt)

I wonder what “restored: true” means?
Anyway, there are sensors without that attribute not working as well.

I had the impression that removing the cameras from the integration and adding them again gets the sensors back – but they still disappear again after a few hours.

**Comment by JurajNyiri** [comment]
> Restarting HA removes sensors that have been working fine before.
> Even triggering the sensors does not bring them back.

Read FAQ.

**Comment by kranzhoff** [comment]
Thanks for getting back on this.
I read the FAQ at least 10 times and I don't see what I'm missing.

Sensors of all 4 cameras did work a couple of times,
therefore network including webhooks should be fine.
There are no disabled entities in .storage/core.entity_registry.

All cameras run the latest firmware version 1.3.9

Meanwhile camera #3 has both sensors back (without me doing anything).
All other cameras have both sensors "unavailable" and one even "missing entity".

**Comment by JurajNyiri** [comment]
> Motion sensor is added only after a motion is detected for the first time.

**Comment by kranzhoff** [comment]
Sure, but as I said before: 

I triggerd motion and noise alarm on all cameras after restarting HA - to no avail.
Camera #1 reports 10+ motion events since the restart, but the sensor is "unavailable":

[binary_sensor.camera_eingang_motion_alarm]
Camera_Eingang Motion Alarm
unavailable	restored: true
device_class: motion
friendly_name: Camera_Eingang Motion Alarm
supported_features: 0

**Comment by JurajNyiri** [comment]
This integration reuses official onvif integration. See if it works there with no issues if you have tried everything else in the faq. 

You can also enable debug logs both for onvif and this integration to see what exactly is going on with onvif.

**Comment by kranzhoff** [comment]
Thanks again for the speedy reply!

The debug log for the Tapo integration shows the same error for all cameras:

> Logger: homeassistant.config_entries
Source: config_entries.py:402
First occurred: 11:30:58 (60 occurrences)
Last logged: 15:53:04
> Error setting up entry 192.168.0.87 for select
Error setting up entry 192.168.0.87 for siren
Error setting up entry 192.168.0.87 for update
Error setting up entry 192.168.0.87 for binary_sensor
Error setting up entry 192.168.0.87 for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 402, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/camera/__init__.py", line 440, in async_setup_entry
    return await component.async_setup_entry(entry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 187, in async_setup_entry
    raise ValueError("Config entry has already been setup!")
ValueError: Config entry has already been setup!

I will dig into the onvif logs now.

**Comment by JurajNyiri** [comment]
That looks like something wrong with the environment.

Try completely removing the cameras including entries for tapo_control inside .storage. 

Then add cameras back and it should be ok.

**Comment by kranzhoff** [comment]
Thanks, I will give that a try.

Available sensors should stay available until a restart of HA, correct?
After a restart they will be unavailable until they are triggered once again, correct?

**Comment by falseunity** [comment]
Same problem here. Everything works for a few days then 1 of the 4 cams shows unavailable for the motion entity. Triggering it, reloading integration, rebooting cam, rebooting HA, nothing brings it back. Removing the cam and re-setting it up again, brings it back working for a few more days. Then start all over again with a different cam unavailable. All C110, All same FW, etc. can't see any reason why...

**Comment by falseunity** [comment]
Also, FYI, installed the ONVIF integration a few days ago and it works without issue. The motion sensors appear and they've never gone unavailable since I installed it.

**Comment by kksligh** [comment]
I’ve tried the removal, including the .storage removal, and even removed the integration completely - still throws this error every restart. I’ve tried the ONVIF integration and it only exposes a fraction of the entities. So that’s not an option either.

**Comment by falseunity** [comment]
I agree with the ONVIF comment, it was just suggested we try it to see if the Motion Sensors do the same thing in there, and from my observations, they don't. But yes, ONVIF has many less entities that the Tapo integration for sure....

**Comment by JurajNyiri** [comment]
Closing as this now looks to be fixed on new firmware. If issue persists, please open new issue.
